pkg/proxy/apis/config/validation/validation.go

```diff
@@ -205,6 +207,22 @@ func validateProxyModeWindows(mode kubeproxyconfig.ProxyMode, fldPath *field.Pat
 	return field.ErrorList{field.Invalid(fldPath.Child("ProxyMode"), string(mode), errMsg)}
 }
 
+func validateDetectLocalMode(mode kubeproxyconfig.LocalMode, fldPath *field.Path) field.ErrorList {
+	validModes := sets.NewString(
+		string(kubeproxyconfig.LocalModeClusterCIDR),
+		string(kubeproxyconfig.LocalModeNodeCIDR),
+		string(kubeproxyconfig.LocalModeBridgeInterface),
+		string(kubeproxyconfig.LocalModeInterfaceNamePrefix),
+	)
+
+	if mode == "" || validModes.Has(string(mode)) {
+		return nil
+	}
+
+	errMsg := fmt.Sprintf("must be %s or blank", strings.Join(validModes.List(), ","))
+	return field.ErrorList{field.Invalid(fldPath, string(mode), errMsg)}
```

Comment:
Actually, shouldn't this be `field.NotSupported()`, which already formats a list of options?

Reply:
ah... I copied from `validateProxyModeLinux`, but that needs a custom error string, which `field.NotSupported()` doesn't... support